src/v2/expressions.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * A CEL expression which can be evaluated during function deployment, and
+ * resolved to a value of the generic type parameter: i.e, you can pass
+ * an Expression<number> as the value of an option that normally accepts numbers.
+ */
+export abstract class Expression<
```

Comment:
In addition to this, I wonder if things would be cleaner if the two following existed

```
export type ExpressionOrType<T> = T | Expression<T>;
export type NullableExpressionOrType<T> = ExpressionOrType<T> | null;
```

The reason I bring this up is because I saw an earlier instance of `boolean | Expression<string>` that may not have been intentional. This hardens the typing slightly.

WDYT?